File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int k = 0;
        int cnt = 0;
        for (int i = 0; i < nums.length; i++) {
            if (i != 0 && nums[i-1] != nums[i]) {
                cnt = 0;
            }
            if (cnt >= 2) {
                continue;
            }
            nums[k++] = nums[i];
            cnt++;
        }
        return k;
    }
}

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let lenWithoutDuplicates = 1;
    let i = 0;
    let j = 1;
    let snowballSize = 0;

    while (j < nums.length) {
        if (nums[i] === nums[j]) {
            if (snowballSize === 0) {
                snowballSize++;
                nums[++i] = nums[j++];
                lenWithoutDuplicates++;
            } else {
                j++;
            }
        } else {
            snowballSize = 0;
            nums[++i] = nums[j++];
            lenWithoutDuplicates++;
        }
    }

    return lenWithoutDuplicates;
};